juju/juju
# Improvements to ingress address watcher

## Description of change

The way the firewaller watches for ingress address changes is altered to use a stringswatcher instead of a notifywatcher. The watcher event now contains the working set of ingress addresses; there is no need for a separate api call to get the addresses, so this facade method is removed.

A new IngressAddressWatcher is added which watches for units on a given relation to enter'leave scope, and returns the set of public addresses for all units in scope. This is not used yet but will be in a subsequent PR.

## QA steps

Run up a CMR scenario on AWS.
Ensure the units can talk to each other.
Check the AWS sec group rules to ensure the expected subnets are used.
